src/ripple/rpc/handlers/LedgerRequest.cpp

```diff
@@ -76,10 +76,19 @@ Json::Value doLedgerRequest (RPC::Context& context)
             {
                 // We don't have the ledger we need to figure out which ledger
                 // they want. Try to get it.
-                Json::Value jvResult
-                        =  getApp().getInboundLedgers().findCreate (
-                            refHash, refIndex, InboundLedger::fcGENERIC)
-                        ->getJson (0);
+
+                auto il = getApp ().getInboundLedgers ().findCreate (
```

Comment:
Not so important comment you don't have to do:

C and C++ have always let you declare a variable inside a `for`, `while` or `if` clause, so I've always used this sort of pattern to good success before:

                if (auto il = getApp ().getInboundLedgers ().findCreate (
                        refHash, refIndex, InboundLedger::fcGENERIC))
                {
                    Json::Value jvResult = il->getJson (0);

                    jvResult[jss::error] = "ledgerNotFound";
                    return jvResult;
                }

                // Is rpcINTERNAL really the best error type for this?
                return RPC::make_error (
                    rpcINTERNAL,
                    "findCreate failed to return an inbound ledger");

The advantage is that the scoping of `il` is more precise, and it uses a couple of lines less code.